microcosm_pubsub/handlers/uri_handler.py

```diff
@@ -125,14 +125,20 @@ def get_resource(self, message, uri):
         Passes message context.
 
         """
-        # XXX we only want X-Request headers
-        headers = self.opaque.as_dict()
+        headers = self.get_headers(message)
         response = get(uri, headers=headers)
         if response.status_code == codes.not_found and self.nack_if_not_found:
             raise Nack(self.resource_nack_timeout)
         response.raise_for_status()
         return response.json()
 
+    def get_headers(self, message):
```

Comment:
Is the signature here just to look like the call in our clients? `message` isn't used here.

Reply:
I am anticipating a subclass wanting to extend this behavior using `message` content.